libs/cropper.js

```diff
@@ -0,0 +1,534 @@
+      this.pointerOffset.xSpan = this.selector.offsetLeft + this.selector.offsetWidth - options.selector.min.width;
+      this.pointerOffset.ySpan = this.selector.offsetTop + this.selector.offsetHeight - options.selector.min.height;
+      this.toggleHandles(false, type);
+      window.onpointerup = onPointerUp;
+      window.onpointermove = onPointerMove;
+    }
+    /**
+     * Generic selector resizing handler.
+     *
+     * @param {boolean} top Flag denoting top handle location.
+     * @param {boolean} left Flag denoting left handle location.
+     * @param {boolean} bottom Flag denoting bottom handle location.
+     * @param {boolean} right Flag denoting right handle location.
+     */
+    const handleAll = (top, left, bottom, right) => {
+      let width;
+      let height;
+      const parentBounds = this.canvas.parentElement.getBoundingClientRect();
+      if (top) {
+        const ySpan = this.selector.offsetTop + this.selector.offsetHeight;
+        let top = event.pageY - parentBounds.top;
+        if (top < 0) {
+          top = 0;
+        }
+        if (top > this.pointerOffset.ySpan) {
+          top = this.pointerOffset.ySpan;
+        }
+        height = ySpan - top;
+        this.selector.style.top = top + 'px';
+      }
+      if (left) {
+        const xSpan = this.selector.offsetLeft + this.selector.offsetWidth;
+        let left = event.pageX - parentBounds.left;
+        if (left < 0) {
+          left = 0;
+        }
+        if (left > this.pointerOffset.xSpan) {
+          left = this.pointerOffset.xSpan;
+        }
+        width = xSpan - left;
+        this.selector.style.left = left + 'px';
+      }
+      if (bottom) {
+        const maxHeight = this.canvas.offsetHeight - this.selector.offsetTop;
+        height = event.pageY - parentBounds.top - this.selector.offsetTop;
+        if (height < 0) {
+          height = 0;
+        }
+        if (this.selector.offsetTop + height > this.canvas.offsetHeight) {
+          height = maxHeight;
+        }
+      }
+      if (right) {
+        const maxWidth = this.canvas.offsetWidth - this.selector.offsetLeft;
+        width = event.pageX - parentBounds.left - this.selector.offsetLeft;
+        if (width < 0) {
+          width = 0;
+        }
+        if (this.selector.offsetLeft + width > this.canvas.offsetWidth) {
+          width = maxWidth;
+        }
+      }
+      if (width) {
+        if (width < options.selector.min.width) {
+          width = options.selector.min.width;
+        }
+        this.selector.style.width = width + 'px';
+      }
+      if (height) {
+        if (height < options.selector.min.height) {
+          height = options.selector.min.height;
+        }
+        this.selector.style.height = height + 'px';
+      }
+    }
+    const onPointerMove = (event) => {
+      event.preventDefault();
+      event.stopPropagation();
+      const map = {
+        t: false,
+        l: false,
+        b: false,
+        r: false
+      }
```

Comment:
Yep. The map uses only letters because the type is a string that consists of those letters.
So if type is the "tl" string that's short for top-left and the code for "t" and "l" will be used. Type is parsed with a for loop to get its letters.
I guess I could update the code to use specific words sent as an object instead of a string.